src/other/index.jl

```diff
@@ -108,19 +108,45 @@ function Base.insert!(x::Index, idx::Integer, nm::Symbol)
     x
 end
 
-Base.getindex(x::Index, idx::Symbol) = x.lookup[idx]
-Base.getindex(x::AbstractIndex, idx::Real) = Int(idx)
-Base.getindex(x::AbstractIndex, idx::AbstractVector{Union{Bool, Missing}}) =
-    getindex(x, collect(Missings.replace(idx, false)))
+Base.getindex(x::AbstractIndex, idx::Symbol) = x.lookup[idx]
+Base.getindex(x::AbstractIndex, idx::AbstractVector{Symbol}) = [x.lookup[i] for i in idx]
+Base.getindex(x::AbstractIndex, idx::Integer) = Int(idx)
+Base.getindex(x::AbstractIndex, idx::AbstractVector{Int}) = idx
+Base.getindex(x::AbstractIndex, idx::AbstractRange) = getindex(x, collect(idx))
+
 function Base.getindex(x::AbstractIndex, idx::AbstractVector{Bool})
     length(x) == length(idx) || throw(BoundsError(x, idx))
     find(idx)
 end
-Base.getindex(x::AbstractIndex, idx::AbstractVector{T}) where {T >: Missing} =
-    getindex(x, collect(skipmissing(idx)))
-Base.getindex(x::AbstractIndex, idx::AbstractRange) = [idx;]
-Base.getindex(x::AbstractIndex, idx::AbstractVector{T}) where {T <: Real} = convert(Vector{Int}, idx)
-Base.getindex(x::AbstractIndex, idx::AbstractVector{Symbol}) = [x.lookup[i] for i in idx]
+
+function Base.getindex(x::AbstractIndex, idx::AbstractVector{<:Integer})
+    # TODO: this line should be changed to throw an error after deprecation
+    if any(v -> v isa Bool, idx)
```

Comment:
You should be able to move this to deprecated.jl too with `::AbstractVector{Bool}`. Also better use `convert` below to avoid making a copy of `idx` when possible.

Reply:
This may not go do deprecated.jl, because - at least as is - it will have to stay here and throw an error if `Bool` is in any entry of `idx`.

Given that above there is a method (line 114):
```
Base.getindex(x::AbstractIndex, idx::AbstractVector{Int}) = idx
```
when do you believe that in this method no conversion would be required?